develop/buildkite/pipeline.yml

```diff
@@ -306,21 +325,39 @@ steps:
           run: integration-test-sqlite
           config: ./develop/buildkite/docker-compose.yml
 
-  - wait
+  - wait: ~
+    continue_on_failure: true  # required for test reports of failed build to be picked up
+  
+  - label: ":mag: test-report"
+    agents:
+      queue: "default"
+      docker: "*"
+    command: buildkite-agent artifact download ".testreport/*.xml" .
+    retry:
+      automatic:
+        limit: 2
+    plugins:
+      - test-collector#v1.10.0:
+          files: ".testreport/*.xml"
+          format: junit
+  
+  - wait # required to prevent uploading partial coverage data for failed builds
```

Comment:
I wish there was a way that this and the coverage report could run in parallel when both run... it's only ~30s but my other PR goes through a bunch of effort to try to reduce those short delays 🫤

Reply:
Sigh. Yeah, I don't like that one bit either and would like them to run in parallel. Here are the tradeoffs:

(a) Run them in parallel. Downside: test coverage would be uploaded for failed builds.

(b) Run them sequentially so that test reporting is always happening, but test coverage is only uploaded when the build was a success. Downside: adds 30+ extra seconds.

Let's change it to (a) since having a broken `main` does happen, but will be rare?

Reply:
FWIW, I sent Buildkite support an email to ask if there's a better solution. But I'd still be in favour of merging it for now.

Reply:
tl;dr of their answer:

> I took a look at this and unfortunately you wouldn't be able to achieve use case this way